buf-api-update/tsconfig.json

```diff
@@ -1,12 +1,9 @@
 {
     "compilerOptions": {
-      "target": "es6",
-      "module": "commonjs",
-      "outDir": "./lib",
-      "rootDir": "./src",
+      "lib": ["ES2023"],
+      "module": "node16",
+      "target": "ES2022",
       "strict": true,
```

Comment:
From https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-20.

Keeping the file as minimal as possible should mean we break things as seldom as possible.